File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        s = s.replaceAll("[^a-zA-Z0-9]", "");
        int end = s.length();
        for(int i=0; i<end/2; i++){
            if(s.charAt(i) != s.charAt(end-1-i)){
                return false;
            }
            
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
import re
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clear_string = re.sub(r'[^a-z0-9]','',s.lower())
        length = len(clear_string)
        
        
        for i in range(length//2):
                if clear_string[i] != clear_string[length-1-i]:
                    return False
        return True
                    
        